feat: added lab to build a quickstart (#1686)
newrelic/developer-website: src/markdown-pages/instant-observability/build-a-quickstart/deploy-your-application.mdx

---
path: '/instant-observability/build-a-quickstart/deploy-your-application'
title: 'Deploy your application'
template: 'GuideTemplate'
description: 'Spin up your demo services to follow along the lab'
duration: '5 min'
---

<Callout variant="course" title="Lab">

> This procedure is a part of lab that teaches you how to build a quickstart. If you haven't already, check out the [_lab introduction_](/instant-observability/build-a-quickstart)

Reviewer: ```suggestion
This procedure is a part of lab that teaches you how to build a quickstart. If you haven't already, check out the [_lab introduction_](/instant-observability/build-a-quickstart).
```

</Callout>

> Before you build a quickstart, you need to spin up demo services. The lab depends on two important services:

Reviewer: ```suggestion
Before you build a quickstart, you need to spin up two demo services:
```

> - A python program that mimics the database and also provides functions for CRUD operations. It also makes use of `newrelic_telemetry_sdk` to send telemetry data to New Relic One.

Reviewer: ```suggestion
- A python program that mimics the database and also provides functions for create, read, update, and delete (CRUD) operations. It also makes use of [newrelic_telemetry_sdk](https://github.com/newrelic/newrelic-telemetry-sdk-python) to send telemetry data to New Relic.
```

- A simulator service that generates dummy database traffic so you don't have to manually perform CRUD operations

> To spin up demo services, you first need to install [Docker](https://www.docker.com/) and also set **NEW_RELIC_INSERT_KEY** environment variable.

Reviewer: ```suggestion
To spin up these services, you first need to install [Docker](https://www.docker.com/) and also set **NEW_RELIC_INSERT_KEY** environment variable.
```

Reviewer: I would add "Install [Docker](https://www.docker.com/)" to the Prerequisites on the intro page.

Reviewer: Setting the environment variable is an actual step, so I'd just make a small section here:

## Prepare your environment

On your local machine, set your license key in an environment variable:

```bash
export NEW_RELIC_INSERT_KEY=<YOUR_LICENSE_KEY>
```

Your mock database uses this key to write telemetry data to New Relic. [You can find your New Relic license key in your account settings](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/)

> <Callout variant='tip'>

Reviewer: You can get rid of this callout if you agree with my suggestion above ^

**NEW_RELIC_INSERT_KEY** environment variable is your New Relic liscence key. [You can find your New Relic liscence key in your New Relic account](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/). 

</Callout>

## Spin up demo services

Clone the lab repository from GitHub: 
```bash

>  git clone https://github.com/mehreentahir16/FlashDB.git

Reviewer: Let's move this under newrelic-experimental instead of your personal GitHub

``` 

This repository contains the code for flashdb as well as the simulator for dummy database traffic. Navigate to the demo directory and execute the following commands to build and run docker image.

>  ```bash

Reviewer: ````
```bash animate
docker build -t flashdb .
docker run flashdb
[output] Writing...
[output] try send
[output] Writing...
[output] try send
[output] Reading...
[output] {green}Sent metrics successfully!
```
````

 docker build -t flashdb . 

>  docker run flashdb

Reviewer: This needs to be:

```docker run -e NEW_RELIC_INSERT_KEY=$NEW_RELIC_INSERT_KEY flashdb```

```

Here, you build and run docker image that generates mock database traffic. You will observe this data in New Relic. Once the docker image is up and running, you see the following output in your terminal:

![deploy-your-application](../../../images/instant-observability/terminal-output.png)

<Callout variant='important'>

> Make sure you set NEW_RELIC_INSERT_KEY environment variable with your actual [license key](https://docs.newrelic.com/docs/accounts/accounts-billing/account-setup/new-relic-license-key/).

Reviewer: If you added this as a section/step above as suggested, you can delete this from here.

</Callout>

> The next step is to observe this data in New Relic. 

Reviewer: A bit impersonal, how about:

> Next, you observe this data in New Relic.
